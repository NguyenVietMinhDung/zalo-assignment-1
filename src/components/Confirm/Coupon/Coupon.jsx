import React from 'react';
import { Container, Row, Col } from 'react-grid-system';
import './Coupon.scss';

const Coupon = () => (
  <Container>
    <Row className="coupon">
      <Col className="coupon__title" xs={12}>
        <p>Thêm mã giảm giá</p>
      </Col>
      <Col className="coupon__content" xs={12} md={6} lg={4}>
        <div className="coupon__content__container">
          <div className="coupon__content__code">
            <i className="lni lni-16 lni-gift" />
            <span>VNGTHANG7</span>
            <i className="lni lni-16 lni-checkmark-circle" />
          </div>
          <div className="coupon__content__btn">
            ÁP DỤNG
          </div>
        </div>
      </Col>
    </Row>
  </Container>
);

export default Coupon;
