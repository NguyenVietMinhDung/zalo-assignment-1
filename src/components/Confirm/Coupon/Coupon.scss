@import '../../../styles/variables';

.coupon {
	display: flex;

	flex-direction: column;

	margin-bottom: 8px;

	padding-bottom: 15px;

	background-color: white;

	&__title {
		font-size: 12px;
		font-weight: 500;
	}

	&__content {
		display: flex;
		align-items: center;
		justify-content: center;

		&__container {
			display: flex;

			width: 400px;
			max-width: 100vw;
			padding: 5px 20px;

			border: 1px solid $dark-white-smoke;
			border-radius: 50px;
			align-items: center;
			justify-content: space-between;
		}

		&__code {
			i:first-child {
				margin-right: 5px;

				color: $light-blue;
			}

			i:last-child {
				margin-left: 5px;

				color: green;
			}
		}

		&__btn {
			padding: 5px 20px;

			cursor: pointer;

			color: white;
			border-radius: 50px;
			background-color: $light-blue;
		}
	}
}
