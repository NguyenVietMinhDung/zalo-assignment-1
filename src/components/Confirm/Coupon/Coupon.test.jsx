import React from 'react';
import { shallow } from 'enzyme';
import Coupon from './Coupon';

describe('<Coupon />', () => {
  test('renders correctly', () => {
    const component = shallow(<Coupon />);
    expect(component).toMatchSnapshot();
  });
});
