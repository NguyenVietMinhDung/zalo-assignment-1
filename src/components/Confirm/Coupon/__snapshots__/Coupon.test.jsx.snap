// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Coupon /> renders correctly 1`] = `
<Container
  component="div"
  fluid={false}
  lg={false}
  md={false}
  sm={false}
  style={Object {}}
  xl={false}
  xs={false}
  xxl={false}
>
  <Row
    align="normal"
    className="coupon"
    component="div"
    debug={false}
    gutterWidth={null}
    justify="start"
    nogutter={false}
    nowrap={false}
    style={Object {}}
  >
    <Col
      className="coupon__title"
      component="div"
      debug={false}
      lg={null}
      md={null}
      offset={Object {}}
      pull={Object {}}
      push={Object {}}
      sm={null}
      style={Object {}}
      width={null}
      xl={null}
      xs={12}
      xxl={null}
    >
      <p>
        Thêm mã giảm giá
      </p>
    </Col>
    <Col
      className="coupon__content"
      component="div"
      debug={false}
      lg={4}
      md={6}
      offset={Object {}}
      pull={Object {}}
      push={Object {}}
      sm={null}
      style={Object {}}
      width={null}
      xl={null}
      xs={12}
      xxl={null}
    >
      <div
        className="coupon__content__container"
      >
        <div
          className="coupon__content__code"
        >
          <i
            className="lni lni-16 lni-gift"
          />
          <span>
            VNGTHANG7
          </span>
          <i
            className="lni lni-16 lni-checkmark-circle"
          />
        </div>
        <div
          className="coupon__content__btn"
        >
          ÁP DỤNG
        </div>
      </div>
    </Col>
  </Row>
</Container>
`;
